pkg/serial/handshake.go

```diff
@@ -40,6 +40,9 @@ const (
 
 // PurgeIncoming is used to clear a channel of bytes prior to handshaking
 func PurgeIncoming(conn net.Conn) {
+	if tracing {
```

Comment:
BTW, you could modify `trace.newTrace`, make it public, and wrap this `if() { trace... }` in function with an appropriate skip (3 looks right).